api/src/main/java/com/messagebird/objects/integrations/HSMComponentType.java

```diff
@@ -23,13 +37,10 @@ public enum HSMComponentType {
 
   @JsonCreator
   public static HSMComponentType forValue(String value) {
-    for (HSMComponentType hsmComponentType : HSMComponentType.values()) {
-      if (hsmComponentType.getType().equals(value)) {
-        return hsmComponentType;
-      }
+    if (value == null) {
```

Comment:
another way to through that exception as well
Objects.requireNonNull(value, "Value cannot be null");  //it will throw that exception as well
return TYPE_MAP.get(value.toLowerCase(Locale.ROOT));

Reply:
updated, thanks